packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1367,15 +1339,161 @@ class RouteObserver<R extends Route<dynamic>> extends NavigatorObserver {
 abstract class RouteAware {
   /// Called when the top route has been popped off, and the current route
   /// shows up.
-  void didPopNext() { }
+  void didPopNext() {}
 
   /// Called when the current route has been pushed.
-  void didPush() { }
+  void didPush() {}
 
   /// Called when the current route has been popped off.
-  void didPop() { }
+  void didPop() {}
 
   /// Called when a new route has been pushed, and the current route is no
   /// longer visible.
-  void didPushNext() { }
+  void didPushNext() {}
+}
+
+class _DialogRoute<T> extends PopupRoute<T> {
```

Comment:
I imagine you can get rid of the _DialogRoute from material as well and just invoke the new show with the right parameters now?